js/src/html/beautifier.js

```diff
@@ -695,7 +695,7 @@ Beautifier.prototype._set_tag_position = function(printer, raw_token, parser_tok
     }
 
     // Don't add a newline before elements that should remain where they are.
-    if (parser_token.tag_name === '!--' && last_token.type === TOKEN.TAG_CLOSE &&
+    if ((parser_token.tag_name !== '!--' || parser_token.tag_name === '!--') && last_token.type === TOKEN.TAG_CLOSE &&
```

Comment:
When does the expression ` (parser_token.tag_name !== '!--' || parser_token.tag_name === '!--')` not evaluate to `true`?   What happens if you remove this expression?

Reply:
It has to deal with the spacing issue that it is reading when reading the comments. It is reading the tag name that does not have a space after <!-- (ex. <!--end comment--> as the tag name is not equal to '!== 'and vice versa. This expression covers both edge cases.

Reply:
How would you like me to add tests?

Reply:
Tests: See #1844.  You only the to update the test data file.  The `generated` files are generated by the build.

Reply:
@carolinekistler
Please fetch and merge the latest changes from main.  I've just removed the generated files from tracking to improve clarity.